mass_mailing_custom_unsubscribe/models/mail_blacklist.py

```diff
@@ -0,0 +1,36 @@
+# Copyright 2019 Tecnativa - Ernesto Tejeda
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import models
+
+
+class MailBlackList(models.Model):
+    _inherit = 'mail.blacklist'
+
+    def _add(self, email):
+        mailing_id = self.env.context.get('mailing_id')
```

Comment:
```suggestion
        mailing_id = self.env.context.get('mailing_id')
        res_id = self.env.context.get('unsubscription_res_id')
```

The `res_id` is required for the whole `if` block to work fine.